app/models/SchemeType.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import play.api.libs.json._
+import utils.{InputOption, WithName}
+
+sealed trait SchemeType
+
+object SchemeType {
+
+  case object SingleTrust extends WithName("singleTrust") with SchemeType
+
+  case object GroupLifeDeath extends WithName("groupLifeDeath") with SchemeType
+
+  case object BodyCorporate extends WithName("bodyCorporate") with SchemeType
+
+  case class Other(schemeTypeDetails: String) extends WithName("other") with SchemeType
+
+  val mappings: Map[String, SchemeType] = Seq(
+    SingleTrust,
+    GroupLifeDeath,
+    BodyCorporate
+  ).map(v => (v.toString, v)).toMap
+
+  def options: Seq[InputOption] = Seq(
+    InputOption(SingleTrust.toString, s"schemeType.type.${SingleTrust.toString}"),
+    InputOption(GroupLifeDeath.toString, s"schemeType.type.${GroupLifeDeath.toString}"),
+    InputOption(BodyCorporate.toString, s"schemeType.type.${BodyCorporate.toString}"),
+    InputOption(Other.toString, s"schemeType.type.${Other.toString}", Some("schemeType_schemeTypeDetails-form"))
```

Comment:
All of the input options will be using their `WithName(...)` values except `Other`. This is because they're all objects but `Other` is a case class. 

As this is within the model, I'd be ok with setting this to `"other"` which would be inline with the JS `Reads`/`Writes` below.